src/panels/logbook/ha-logbook.ts

```diff
@@ -25,20 +25,15 @@ class HaLogbook extends LitElement {
   // @ts-ignore
   private _rtl = false;
 
-  protected updated(changedProps: PropertyValues) {
-    super.updated(changedProps);
-    if (!changedProps.has("hass")) {
-      return;
-    }
-    const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
-    if (oldHass && oldHass.language !== this.hass.language) {
-      this._rtl = computeRTL(this.hass);
+  protected shouldUpdate(changedProps: PropertyValues) {
+    const oldRtl = this._rtl;
+    if (changedProps.has("hass")) {
+      const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
+      if (oldHass && oldHass.language !== this.hass.language) {
+        this._rtl = computeRTL(this.hass);
```

Comment:
We shouldn't have side effects in our `shouldUpdate`. Setting `this._rtl` should stay in `updated`.

Reply:
we don't evne use `this._rtl` it seems? We can drop it.

Reply:
It is used, reflected attribute

Reply:
`shouldUpdate` should return true when the language is changed or when `entries` is changed.

This change will affect state icons btw, they will no longer be updated when a state changed. But I don't think that is a big problem as the icon would not match the logbook entry anyway? 🤔

Reply:
Yes, `rtl` is used for date title. My idea was calculate `this._rtl` before the update, so I think `performUpdate` is a better place than `updated`.

Reply:
No need to do it in `performUpdate` as your shouldUpdate is false when `_rtl` is changed, it will not cause another render.